pages/api/applications/[applicationId]/tasks/[taskId].ts

```diff
@@ -1,5 +1,116 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse, StatusMessageType } from '../../../../../types/apiResponse';
+import { TaskData, TaskPatchData } from '../../../../../types/task';
+import { withAuthUser } from '../../../../../utils/auth/jwtHelpers';
+import { isValidDate } from '../../../../../utils/date/validations';
+import { createJsonResponse, HttpMethod, HttpStatus, rejectHttpMethod } from '../../../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../../../utils/prisma/prismaHelpers';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const prisma = new PrismaClient();
+
+enum MessageType {
+  INVALID_TASK_ID,
+  TASK_NOT_FOUND,
+  INVALID_APPLICATION_ID,
+  TASK_DELETED,
+  TASK_PATCHED,
+  INVALID_TITLE,
+  INVALID_DUE_DATE,
+  INVALID_NOTIFICATION_DATETIME,
+  INVALID_IS_DONE,
+}
+
+const messages = Object.freeze({
+  [MessageType.INVALID_TASK_ID]: { type: StatusMessageType.ERROR, message: 'Task id is invalid.' },
+  [MessageType.INVALID_APPLICATION_ID]: { type: StatusMessageType.ERROR, message: 'Application id is invalid.' },
+  [MessageType.TASK_NOT_FOUND]: { type: StatusMessageType.ERROR, message: 'Task cannot be found.' },
+  [MessageType.TASK_DELETED]: { type: StatusMessageType.SUCCESS, message: 'The task has been successfully deleted.' },
+  [MessageType.TASK_PATCHED]: { type: StatusMessageType.SUCCESS, message: 'The task has been successfully patched.' },
+  [MessageType.INVALID_TITLE]: { type: StatusMessageType.ERROR, message: 'The task title is invalid.' },
+  [MessageType.INVALID_DUE_DATE]: { type: StatusMessageType.ERROR, message: 'The task due date is invalid.' },
+  [MessageType.INVALID_NOTIFICATION_DATETIME]: {
+    type: StatusMessageType.ERROR,
+    message: 'The task notification date is invalid.',
+  },
+  [MessageType.INVALID_IS_DONE]: { type: StatusMessageType.ERROR, message: 'The task completion status is invalid.' },
+});
+
+async function handler(uid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<TaskData>>) {
+  const { taskId, applicationId } = req.query as { taskId: string; applicationId: string };
+
+  if (!Number.isInteger(Number(taskId))) {
+    return res.status(HttpStatus.BAD_REQUEST).json(createJsonResponse({}, messages[MessageType.INVALID_TASK_ID]));
+  }
+
+  if (!Number.isInteger(Number(applicationId))) {
+    return res
+      .status(HttpStatus.BAD_REQUEST)
+      .json(createJsonResponse({}, messages[MessageType.INVALID_APPLICATION_ID]));
+  }
+
+  const task = await prisma.task.findUnique({
+    where: { id: Number(taskId) },
+    include: { application: { select: { id: true, userId: true } } },
+  });
+  if (!task || task.application.id !== Number(applicationId) || task.application.userId !== uid) {
+    return res.status(HttpStatus.NOT_FOUND).json(createJsonResponse({}, messages[MessageType.TASK_NOT_FOUND]));
+  }
```

Comment:
Do we want to use this level of abstraction here? I guess it reduces code duplication in some ways, but it also breaks the current level of abstraction and increases coupling?

Not sure if it's right to put the responsibility of checking `applicationId` or uniqueness or that the task belongs to the user in the top level handler too because what if other methods/apis route to this endpoint in future and don't have the same constraints?

These checks are implicitly handled here so when I look at the `deleteHandler` for example, I started to wonder if we already checked that the user is authorized to delete the task.